### PR TITLE
implement /secret-key endpoints, add stub routes

### PR DESCRIPTION
**Pull request by chuckadams**
# Pull Request

## What changed?

* replaces the single catch-all route with ones that specifically match the wp.org api endpoints and versions
* implements the /secret-key and /secret-key/1.1/salt endpoints
* adds spatie/laravel-data which will be used later
* adds thecodingmachine/safe and a phpstan rule to enforce its use
* adds roave/security-advisories
* Adds a "Local API" environment for bruno tests, fixes all .bru files to use the protocol in the API_URL variable
* adds some Model classes that still need filling out.
 
## Why did it change?

the yaks must be shaved

## Did you fix any specific issues?

none

## CERTIFICATION

By opening this pull request, I do agree to abide by
the [CODE OF CONDUCT](https://github.com/aspirepress/.github/CODE_OF_CONDUCT.md) and be bound by the terms
of the [Contribution Guidelines](https://github.com/aspirepress/.github/CONTRIBUTING.md) in effect on the date and time
of my contribution as proven by the
revision information in GitHub. I also agree that any previous contributions shall be deemed subject to the terms of the
version in effect on the date and time of this pull request, or any future revisions for pull requests I may submit.
Further, I certify that this work is my own, is original, does not violate the intellectual property of any other person
or entity, and I am not violating any license agreements or contracts I have with any person or entity. Finally, I agree
that this code may be licensed under any license deemed appropraite by AspirePress, including but not
limited to open source, closed source, proprietary or custom licenses, and that such license terms neither violate my
rights or my copyright to this code.

### CODE REVIEW

**Comment by asirota** [comment]
Bruno changes look good!

File: app/Http/Controllers/API/WpOrg/SecretKey/SecretKeyController.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    private function generate_1_1(): string
+    {
+        $out = '';
+        foreach (['AUTH_KEY', 'SECURE_AUTH_KEY', 'LOGGED_IN_KEY', 'NONCE_KEY'] as $name) {
+            $param = "'$name',";
+            $out .= sprintf("define(%-18s '%s');\n", $param, self::unique_key());
+        }
+        return $out;
+        // define('AUTH_KEY',        'y*k]cp10`=Ut@0G9GwLD]nYX-#L_dkFyQ=7ts{,=R<T;0N/KmQ>+@Z!Up|5PQIp9');
```
Comment (sarah-savage) [inline app/Http/Controllers/API/WpOrg/SecretKey/SecretKeyController.php:47]: If we can take the commented stubs out, that would be ideal.

File: bruno/Core/Core Stability Check.bru [hunk 2]
```diff
@@ -5,7 +5,7 @@ meta {
 }
 
 get {
-  url: https://{{API_URL}}/core/stable-check//:api_version
+  url: {{API_URL}}/core/stable-check//:api_version
```
Comment (sarah-savage) [inline bruno/Core/Core Stability Check.bru:8]: Do we need the double slash here?

**Review by sarah-savage — APPROVED** [review]
Couple nits here, but I think this is good.

File: bruno/Core/Core Stability Check.bru (hunk 2 above)
Comment (asirota) [reply bruno/Core/Core Stability Check.bru:8]: No that must be a typo. Let me see if I can adjust.

**Pull request merged**